zcash_primitives/src/transaction/mod.rs

```diff
@@ -371,6 +362,54 @@ impl PartialEq for Transaction {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum OrchardBundle<A: orchard::bundle::Authorization> {
+    OrchardVanilla(Box<Bundle<A, Amount, OrchardVanilla>>),
+    #[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
+    OrchardZSA(Box<Bundle<A, Amount, OrchardZSA>>),
+}
+
+/// Errors that can occur during transaction construction.
+#[derive(Debug)]
+pub enum BundleError {
+    WrongBundleType,
+}
```

Comment:
We need to use this every time there is typemismatch. But curently this is not the case. It is not used. Remove or refactor to use it.

Reply:
Removed